spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/CorrelationData.java

```diff
@@ -70,10 +75,44 @@ public void setId(String id) {
 		this.id = id;
 	}
 
+	/**
+	 * Return a future to check the success/failure of the publish operation.
+	 * @return the future.
+	 * @since 2.1
+	 */
 	public SettableListenableFuture<Confirm> getFuture() {
 		return this.future;
 	}
 
+	public String getReturnCorrelation() {
+		return this.returnCorrelation;
+	}
+
+	/**
+	 * Set a unique return correlation id, used to populate the
+	 * {@link #setReturnedmessage(Message) returnedMessage}.
+	 * @param returnCorrelation the correlation id.
+	 * @since 2.1
+	 */
+	public void setReturnCorrelation(String returnCorrelation) {
+		this.returnCorrelation = returnCorrelation;
+	}
+
+	/**
+	 * Return a returned message, if any; requires a {@link #setReturnCorrelation(String)
+	 * returnCorrelation}. Guaranteed to be populated before the future is set.
+	 * @return the message or null.
+	 * @since 2.1
+	 */
+	@Nullable
+	public Message getReturnedmessage() {
+		return this.returnedmessage;
+	}
+
+	public void setReturnedmessage(Message returnedmessage) {
```

Comment:
Shouldn't this be package protected do not let end-user to override?
Or is it harmless from the Framework perspective and that's already end-user responsibility for the damage (s)he does for their application?